File: codes/easy/728 Self Dividing Numbers/discussion.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> result;
        
        for(int i = left, temp = 0; i <= right; i++) {
            for(temp = i; temp > 0; temp /= 10) {
                if(temp % 10 == 0 || i % (temp % 10) != 0) break;
            }

            if(!temp) result.push_back(i);
        }
        return result;
    }
};

File: codes/easy/728 Self Dividing Numbers/discussion.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */
var selfDividingNumbers = function(left, right) {
    const result = [];

    for(let i = left, j = 0; i <= right; i++) {
        for(j = i; j > 0; j = Math.floor(j / 10)) {
            if(j % 10 == 0 || i % (j % 10) != 0) break;
        }
        if(!j) result.push(i);
    }

    return result;
};